src/api/app/controllers/webui/distributions_controller.rb

```diff
@@ -0,0 +1,54 @@
+class Webui::DistributionsController < Webui::WebuiController
+  before_action :set_project
+  after_action :verify_authorized
+
+  # GET /projects/:project/distributions/new
+  def new
+    authorize @project, :update?
+
+    @distributions = Distribution.all.group_by(&:vendor)
+    return if @distributions.present?
+
+    if User.admin_session?
+      redirect_to(new_interconnect_path,
+                  alert: 'There are no distributions configured. Maybe you want to connect to one of the public OBS instances?')
+    else
+      redirect_to(project_repositories_path(project: @project))
+    end
+  end
+
+  # PATCH /projects/:project_name/distributions/:id/toggle
+  def toggle
+    authorize @project, :update?
+    @distribution = Distribution.find(params[:distribution])
```

Comment:
are we super confident that this will never fail? Should we use `find!` or check if `@distribution.present?` ?

Reply:
`find` always raises. 

```ruby
Distribution.find(1312)
ActiveRecord::RecordNotFound: Couldn't find Distribution with 'id'=1312
```